src/services/sync/strategies/WebUntis/WebUntisSchoolyearSyncer.js

```diff
@@ -0,0 +1,544 @@
+	*/
+	async logout(session) {
+		await super.logout();
+	}
+
+	/**
+	* 
+	* @param {*} metadata 
+	*/
+	async storeMetadata(metaData, params) {
+		const webUntisMetadataService = this.app.service('webuntisMetadata');
+
+		await Promise.all(metaData.map(async (entry) => {
+			const metadataResults = await webUntisMetadataService.find({ query: {
+				datasourceId: params.datasourceId,
+				teacher: entry.teacher,
+				class: entry.class,
+				subject: entry.subject
+			}, paginate: false });
+			const result = metadataResults[0];
+
+			if (result !== undefined) { // patch/replace existing
+				await webUntisMetadataService.patch(result._id, { state: entry.state, times: entry.times });
+			} else { // create new
+				await webUntisMetadataService.create(Object.assign(
+					{ datasourceId: params.datasourceId },
+					entry
+				));
+			}
+		}));
+	}
+
+	/**
+	* 
+	* @param {*} data 
+	*/
+	async createCourses(config, data, params) {
+		// Convert metadata to actual schulcloud courses
+		// And reflect changes in metadata store
+		const webUntisMetadataService = this.app.service('webuntisMetadata');
+
+		const importCondition = this.getImportConditionEvaluator(params);
+
+		for (let entry of data) {
+			if (importCondition(entry._id.toString())) {
+				await this.obtainAndUpdateCourseAndClass(config, entry)
+					.then(() => webUntisMetadataService.patch(entry._id, { state: 'imported' }))
+					// .catch(() => webUntisMetadataService.patch(entry._id, { state: 'errored' }));
+					;
+			} else {
+				await webUntisMetadataService.patch(entry._id, { state: 'discarded' });
+			}
+		}
+	}
+
+	/**
+	* Entry: {
+	*    datasourceId: this.data.datasourceId,
+	*    teacher: 'Renz',
+	*    class: '2a',
+	*    times: [ { weekDay, startTime, endTime, room }],
+	*    subject: 'mathe',
+	*    state: 'new',
+	* }
+	*/
+	async obtainAndUpdateCourseAndClass(config, entry) {
+		const courseService = this.app.service('courses');
+		const classService = this.app.service('classes');
+
+		const user = await this.getUser();
+
+		const school = await this.getSchool(user);
+
+		/**
+        * Mapping:
+        *
+        * Schul-Cloud: class, course (per class), lesson
+        * WebUntis: class, subject
+        * German: Klasse, Kurs, Fach, Schulstunde
+        */
+
+		// Get class
+		const scClasses = await classService.find({ query: { name: entry.class }, paginate: false });
+
+		let scClass = scClasses.length >= 1 ? scClasses[0] : undefined;
+		if (scClass === undefined) {
+			// Create Schul-Cloud class?
+
+			// TODO: extract gradeLevel
+
+			const newClass = {
+				name: entry.class,
+				schoolId: school._id,
+				nameFormat: 'static',
+				year: school.currentYear,
+			};
+			scClass = await classService.create(newClass);
+
+			// TODO: derive successor value for predecessor class?
+		}
+
+		// Extract class short name
+		const shortClassName = scClass.name.match(/[A-Za-z]{1}[0-9]{1,2}[A-Za-z]{1}/g) || scClass.name.substr(0, 3).replace( /^\s+|\s+$/g, '' );
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [There should be no space before this paren.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4061937)